plugins/aggregators/valuecounter/valuecounter.go

```diff
@@ -0,0 +1,94 @@
+package valuecounter
+
+import (
+	"fmt"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/aggregators"
+)
+
+type aggregate struct {
+	name       string
+	tags       map[string]string
+	fieldCount map[string]int
+}
+
+type ValueCounter struct {
+	cache       map[uint64]aggregate
+	Field_names []string
```

Comment:
Call this `Fields`, which will match with some other aggregator work.

Strictly speaking, we could probably use the normal filtering options, since this is essentially fieldpass, but I think it is important to require this to be opt in.